zone/zone_loot.cpp

```diff
@@ -0,0 +1,198 @@
+#include <vector>
+#include "zone.h"
+#include "../common/repositories/loottable_repository.h"
+#include "../common/repositories/loottable_entries_repository.h"
+#include "../common/repositories/lootdrop_repository.h"
+#include "../common/repositories/lootdrop_entries_repository.h"
+
+void Zone::LoadLootTables(const std::vector<uint32> &loottable_ids)
+{
+	// check if table is already loaded
+	for (const auto &e: loottable_ids) {
+		for (const auto &f: m_loottables) {
+			if (e == f.id) {
+				LogLootDetail("Loottable [{}] already loaded", e);
+				return;
+			}
+		}
+	}
+
+	if (loottable_ids.empty()) {
+		LogLootDetail("No loottables to load");
+		return;
+	}
+
+	auto loottables = LoottableRepository::GetWhere(
+		content_db,
+		fmt::format(
+			"id IN ({})",
+			Strings::Join(loottable_ids, ",")
+		)
+	);
+
+	auto loottable_entries = LoottableEntriesRepository::GetWhere(
+		content_db,
+		fmt::format(
+			"loottable_id IN ({})",
+			Strings::Join(loottable_ids, ",")
+		)
+	);
+
+	std::vector<uint32> lootdrop_ids;
+	for (const auto     &e: loottable_entries) {
+		if (std::find(
+			lootdrop_ids.begin(),
+			lootdrop_ids.end(),
+			e.lootdrop_id
+		) == lootdrop_ids.end()) {
+			lootdrop_ids.push_back(e.lootdrop_id);
+		}
+	}
+
+	if (lootdrop_ids.empty()) {
+		LogLoot("No lootdrops to load for loottable(s) [{}]", Strings::Join(loottable_ids, ","));
+		return;
+	}
+
+	auto lootdrops = LootdropRepository::GetWhere(
+		content_db,
+		fmt::format(
+			"id IN ({})",
+			Strings::Join(lootdrop_ids, ",")
+		)
+	);
+
+	auto lootdrop_entries = LootdropEntriesRepository::GetWhere(
+		content_db,
+		fmt::format(
+			"lootdrop_id IN ({})",
+			Strings::Join(lootdrop_ids, ",")
+		)
+	);
+
+	// emplace back tables to m_loottables if not exists
+	for (const auto &e: loottables) {
+		bool has_table = false;
+
+		for (const auto &l: m_loottables) {
+			if (e.id == l.id) {
+				has_table = true;
+				break;
+			}
+		}
+		if (!has_table) {
+			// add loottable
+			m_loottables.emplace_back(e);
+
+			// add loottable entries
+			for (const auto &f: loottable_entries) {
+				if (e.id == f.loottable_id) {
+					m_loottable_entries.emplace_back(f);
+
+					// add lootdrop
+					for (const auto &g: lootdrops) {
+						if (f.lootdrop_id == g.id) {
+							m_lootdrops.emplace_back(g);
+
+							// add lootdrop entries
+							for (const auto &h: lootdrop_entries) {
+								if (g.id == h.lootdrop_id) {
+									m_lootdrop_entries.emplace_back(h);
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	if (loottable_ids.size() > 1) {
+		LogLoot("Loaded [{}] loottables", m_loottables.size());
```

Comment:
Can we add a timer elapsed to this log

Reply:
Sure thing. It’s fairly lightweight with this mostly happening in zone boot

Reply:
```
  Zone |    Info    | LoadLootTables Loaded [56] loottables (0.000302s)
```